<a href="https://colab.research.google.com/github/starkjones/Food-Sales-Predictions-/blob/main/Project_2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## PROJECT 2: TBD
###22.05.05

Jonathan Jones

In [ ]:
import pandas as pd
import numpy as np

from google.colab import drive
drive.mount('/content/drive')



In [16]:
filename = '/content/drive/MyDrive/Colab Notebooks/Week 7/Memphis TN Residential RE data.xlsx'
df = pd.read_excel(filename)
df.head()

Region Month of Period End  Median Sale Price  \
0  Memphis, TN - 35th Ward Civic Club            May 2013            13000.0   
1                                 NaN           June 2013            13000.0   
2                                 NaN           July 2013            13000.0   
3                                 NaN          April 2014             3500.0   
4                                 NaN            May 2014             4750.0   

   Median Sale Price MoM   Median Sale Price YoY   Homes Sold  \
0                     NaN                     NaN           1   
1                0.000000                     NaN           1   
2                0.000000                     NaN           1   
3                     NaN                     NaN           1   
4                0.357143               -0.634615           2   

   Homes Sold MoM   Homes Sold YoY   New Listings  New Listings MoM   \
0              NaN              NaN           NaN                NaN   
1              0.0              NaN           NaN                NaN   
2              0.0              NaN           NaN                NaN   
3              NaN              NaN           1.0                NaN   
4              1.0              1.0           NaN                NaN   

   New Listings YoY   Inventory  Inventory MoM    Inventory YoY   \
0                NaN        NaN             NaN              NaN   
1                NaN        NaN             NaN              NaN   
2                NaN        NaN             NaN              NaN   
3                NaN        NaN             NaN              NaN   
4                NaN        NaN             NaN              NaN   

   Days on Market  Days on Market MoM  Days on Market YoY  \
0             NaN                 NaN                 NaN   
1             NaN                 NaN                 NaN   
2             NaN                 NaN                 NaN   
3             NaN                 NaN                 NaN   
4             NaN                 NaN                 NaN   

   Average Sale To List  Average Sale To List MoM   Average Sale To List YoY   
0              0.866667                        NaN                        NaN  
1              0.866667                    0.00000                        NaN  
2              0.866667                    0.00000                        NaN  
3              1.000000                        NaN                        NaN  
4              0.968750                   -0.03125                   0.102083

In [19]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 23973 entries, 0 to 23972
Data columns (total 20 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Region                     227 non-null    object 
 1   Month of Period End        23973 non-null  object 
 2   Median Sale Price          23973 non-null  float64
 3   Median Sale Price MoM      23093 non-null  float64
 4   Median Sale Price YoY      22017 non-null  float64
 5   Homes Sold                 23973 non-null  int64  
 6   Homes Sold MoM             23093 non-null  float64
 7   Homes Sold YoY             22017 non-null  float64
 8   New Listings               22972 non-null  float64
 9   New Listings MoM           21909 non-null  float64
 10  New Listings YoY           20793 non-null  float64
 11  Inventory                  15615 non-null  float64
 12  Inventory MoM              14645 non-null  float64
 13   Inventory YoY             12153 non-null  flo

In [20]:
df.duplicated().sum()

109

1) Source of data
  https://www.redfin.com/news/data-center/

2) Brief description of data
  Housing market data for the city of Memphis, TN and surrounding area

3) What is the target?

  Predicting the Days on the Market.

4) Is this a classification or regression problem?
  This is regression problem as the target value is not limited to a finite number of classes or values. 

5) How many features?

  20-2(target & region) columns = 18 features. 

6) How many rows of data.

  23973

7) What, if any, challenges do you foresee in cleaning, exploring, or modeling with this dataset?

  I can already see that there are multiple duplicated rows (109) and numerous missing entries. I am concerned that there may not be enough data to build an accurate model. 

In [22]:
filename2 = '/content/drive/MyDrive/Colab Notebooks/Week 7/Orange County FL_ Housing Data.xlsx'
df2= pd.read_excel(filename2)
df2.head()

Month of Period End  Median Sale Price  Median Sale Price MoM   \
0       February 2012           198000.0                0.389474   
1          March 2012           110000.0               -0.444444   
2          April 2012            71000.0               -0.354545   
3            May 2012            92000.0                0.295775   
4           June 2012            72600.0               -0.210870   

   Median Sale Price YoY   Homes Sold  Homes Sold MoM   Homes Sold YoY   \
0                1.134771           2         1.000000        -0.333333   
1               -0.043478           7         2.500000        -0.222222   
2                0.028986           1        -0.857143        -0.857143   
3                0.022222           7         6.000000        -0.363636   
4               -0.308571           5        -0.285714         0.000000   

   New Listings  New Listings MoM   New Listings YoY   ...  Inventory MoM   \
0            10           0.250000           0.666667  ...        0.075000   
1             7          -0.300000          -0.666667  ...       -0.093023   
2             9           0.285714           2.000000  ...        0.051282   
3             5          -0.444444           0.250000  ...       -0.121951   
4             8           0.600000           0.600000  ...        0.027778   

    Inventory YoY   Days on Market  Days on Market MoM  Days on Market YoY  \
0        -0.203704             227                  88                 143   
1        -0.303571             105                -122                 -39   
2        -0.145833             266                 161                 237   
3        -0.121951              95                -171                 -11   
4        -0.051282              54                 -41                 -38   

   Average Sale To List  Average Sale To List MoM   Average Sale To List YoY   \
0              0.922276                   0.171881                  -0.078961   
1              0.923929                   0.001653                   0.048463   
2              0.923277                  -0.000652                  -0.030794   
3              0.902572                  -0.020705                   0.085996   
4              0.846267                  -0.056305                   0.016855   

   Unnamed: 19  Fed Funds Effective Rate  
0          NaN                      0.09  
1          NaN                      0.12  
2          NaN                      0.11  
3          NaN                      0.11  
4          NaN                      0.11  

[5 rows x 21 columns]

In [24]:
df2.shape

(122, 21)

In [26]:
df2.duplicated().sum()

0

This is a similar dataset centered around Orange County Florida where the city of Orlando is. 

1) Source of data
  https://www.redfin.com/news/data-center/

2) Brief description of data
  Housing market data for the city of Orlando, FL and surrounding area

3) What is the target?

  Predicting the Days on the Market.

4) Is this a classification or regression problem?

  This is regression problem as the target value is not limited to a finite number of classes or values. 

5) How many features?

  21-1(target) columns = 20 features. 

6) How many rows of data.

  122

7) What, if any, challenges do you foresee in cleaning, exploring, or modeling with this dataset?

  There is not enough data in this set. 122 rows would be ok for visual analysis but I anticipate that there would be significant issues with fitting a model. 


  I am still searching for a viable dataset. County and city websites for regions that I am interested in have condensed data sets. An MLS database would be ideal, so that can control the numbers of rows and columns that go into the dataframe. 